Core/System/Info.hs

```diff
@@ -65,16 +65,46 @@ instance IsString OS where
 os :: OS
 os = fromString System.Info.os
 
+-- | Enumeration of the known GHC supported architecture.
+--
+data Arch
+    = I386
+    | X86_64
+    | PowerPC
+    | PowerPC64
+    | PowerPC64LE
+    | Sparc
+    | Sparc64
+    | ARM
+    | AArch64
```

Comment:
I'm not sure about the specific here.

I would call `AArch64` : `ARM64`, and get rid of PowerPC64LE (as this is fundamentally not an architecture). it should be the endian info that should be available someplace close that give this information